packages/core/src/editor/actions.ts

```diff
@@ -54,42 +94,43 @@ export const Actions = (
     /**
      * Add a new Node(s) to the editor.
      *
-     * @param nodes
+     * @param tree
      * @param parentId
+     * @param id
      */
-    add(nodes: Node[] | Node, parentId: NodeId) {
+    addLinkedNodeFromTree(tree: NodeTree, parentId: NodeId, id?: string) {
       const parent = getParentAndValidate(parentId);
-
-      if (!parent.data.nodes) {
-        parent.data.nodes = [];
+      if (!parent.data.linkedNodes) {
+        parent.data.linkedNodes = {};
       }
 
-      if (parent.data.props.children) {
-        delete parent.data.props["children"];
-      }
+      parent.data.linkedNodes[id] = tree.rootNodeId;
```

Comment:
what what what
Can you update the description to mention why we need this?

Reply:
Take a look at the `<Element />` component. We get the NodeTree from the React Element, and create a new Node. We attach the new Node to the parent Node's `linkedNodes` property.